### DISCUSSION TITLE
Reducing False Positives

### DISCUSSION THREAD

**Discussion by gcortes** [blakeblackshear/frigate#3136] [Ask A Question]
I've cut down on the number of false positives for my drive way and else where, but I still get some confusing ones. Here is one of my recent captures:

<img width="517" alt="2022-04-24_18-00-33" src="https://user-images.githubusercontent.com/5271010/165005204-9b41711d-ef13-4952-9750-6ab1116a5e6e.png">

She is well out of my trigger zone as I understand it. This is the layout:

<img width="638" alt="2022-04-24_18-04-57" src="https://user-images.githubusercontent.com/5271010/165005287-7e7e010d-a8d0-4ecf-94bb-e136011ef5d3.png">

This my configuration for the camera in question:

<img width="764" alt="2022-04-24_18-10-05" src="https://user-images.githubusercontent.com/5271010/165005350-41159ad9-850a-4b34-b6ed-aa9928e57529.png">

Is the set up deficient in some way or are false positives just something I'll have to live with? There's a lot of traffic on the sidewalk in front of the driveway. I seldom get triggers for on that motion.

I'm using the CPU for detection. A Coral is on order.

**Comment by ve6rah** [comment]
From what I can tell, the zones and masks only affect the initial detection. Once detection starts, it seems to continue detection on the whole image frame, even outside of your zones.
So for example, if she was in your trigger zone, and then walked out, but remained in the frame, the event continues. (alternatively someone else triggers an event in your trigger zone, like you checking the mailbox, but because she's out there, detection continues)
Now I am FAR from being an expert on Frigate, so my understanding may be incorrect, but those are my observations so far.

**Reply by NickM-27** [reply]
This is partly correct. Zones limit when an event is added to the event page, so if an object enters a zone it will be tracked if it leaves. Masks are ALWAYS applied / considered despite the state of the object.

**Reply by ve6rah** [reply]
hmmm... that doesn't seem consistent with the issue I posted in the other thread where I have bounding boxes showing for items completely within a masked area.

**Reply by NickM-27** [reply]
it depends what kind of mask you are talking about, but with [this issue](https://github.com/blakeblackshear/frigate/discussions/3204) it is consistent. Motion can occur partially outside the mask and the object will be detected within the motion mask (since the motion mask doesn't mask out objects).
